src/sisl/io/vasp/stdout.py

```diff
@@ -55,8 +57,10 @@ def accuracy_reached(self):
         """ True if the line "reached required accuracy" was found. """
         return self.step_to("reached required accuracy")[0]
 
+    @SileBinder()
     @sile_fh_open()
-    def read_energy(self, all=False):
+    @deprecation("all keyword is deprecated, use read_energy[:]() instead", from_version="0.14")
```

Comment:
I would propose:
```python
@deprecate_argument("all", None, <message>, from_version="0.14")
@deprecate("direct calls will no longer return the last entry!", from_version="0.14")
```
or similar, that should produce the same behaviour, and no `warn` in the method would be nice.